### DISCUSSION TITLE
Sugestion: add desktop sharing feature

### DISCUSSION THREAD

**Discussion by HotDrive69** [cockpit-project/cockpit#17237] [Ideas]
I use Cockpit for almost everything in my systems. Stopped using Putty because Cockpit has its own terminal. But, sometimes, I really need to access the desktop. It would be great if Cockpit had some feature for desktop sharing, like built in VNC.

**Comment by garrett** [comment]
Cockpit already has sharing of VNC and SPICE for VMs.

Here's the Red Hat docs on how to access this from within Cockpit: https://access.redhat.com/documentation/en-us/red_hat_hyperconverged_infrastructure_for_virtualization/1.5/html/managing_virtual_machines_using_cockpit/assembly-managing-vms-using-cockpit#task-cockpit-ovirt-machines-machine-accessing_vm (While it's Red Hat docs for Red Hat Enterprise Linux, it should be consistent across distros.)

As Cockpit is intended for servers, it doesn't have the capability to configure desktop sharing for the host system, however.

You _can_, however, set up TigerVNC as a systemd service that can be managed (start/stop/restart) from Cockpit's services page. Here's how to set up TigerVNC on Fedora, for example: https://docs.fedoraproject.org/en-US/fedora/rawhide/system-administrators-guide/infrastructure-services/TigerVNC/ (It's probably similar on CentOS, but different on Ubuntu. I'm sure there are guides for the various distros, depending on what you're doing.)

And, since you mentiond Putty, which is a Windows SSH client, it's probably noteworthy that you can also set up RDP on Linux using xrdp. (The service can be started/stopped/restarted in Cockpit's services page too, of course.)

I hope this helps!

**Reply by HotDrive69** [reply]
Thanks Garrett, sure does.

**Comment by nekohayo** [comment]
I was about to suggest roughly the same thing, but specifically about adding the ability to preconfigure a system user's "Remote Login" headless GNOME-remote-desktop RDP feature, as can be found in gnome-control-center:

<img width="1049" height="239" alt="image" src="https://github.com/user-attachments/assets/9e156462-dd26-4b8f-a7ef-8a99bf45bf24" />

…i.e. being able to create a server that serves for a bunch of remote desktop sessions, headless only, without having to plug a monitor into the server to log into their user sessions just to flip on that switch and configure the credentials/settings for that "Remote Login" session.

Do you think this part would be in scope for Cockpit? To me, _that_ would seem like a killer feature.

**Comment by Daskan** [comment]
A desktop sharing feature would really be appreciated.
Creating a Fedora Server on a Proxmox Host and then serving multiple Desktop Sessions my Users can individually connect to via the WebUI would make life in many aspects much easier.
Them not having to rely on external rdp/vnc tools .. me serving different Desktop Sessions with different virtual hardware requirements .. not having to provide nested VMs which may have different kinds of penalties ..

I know cockpit descriptions says its for servers .. but i dont see much of a difference to "workstations" and also wonder why it should only focus on "servers"?
